### PR TITLE
(enhancement) add test for `visit-note,order-basket & clinical-form` action buttons

### PR DESCRIPTION
**Pull request by donaldkibet**
## Requirements

- [ ] This PR has a title that briefly describes the work done, including the ticket number if there is a ticket.
- [x] My work conforms to the [OpenMRS 3.0 Styleguide](https://om.rs/styleguide).
- [x] My work includes tests or is validated by existing tests.

## Summary
1. This is a follow-up PR to add tests for `action buttons`
2. Cleaned up `action menu.component.tsx` by removing unused code

### CODE REVIEW

File: packages/esm-patient-forms-app/src/clinical-form-action-button.test.tsx [hunk 1]
```diff
@@ -0,0 +1,24 @@
...
+  it('should display clinical form action button on tablet view', () => {
+    spyOn(mockEsmFramework, 'useLayoutType').and.returnValue('tablet');
+    render(<ClinicalFormActionButton />);
+    const clinicalActionButton = screen.getByRole('button', { name: /Clinical form/ });
+
+    expect(clinicalActionButton).toBeInTheDocument();
+    expect(clinicalActionButton).toHaveClass('container');
+  });
+
+  it('should display clinical form action button on tablet view', () => {
```
Comment (nanfuka) [inline packages/esm-patient-forms-app/src/clinical-form-action-button.test.tsx:16]: line 16 looks like a repetition of line 7

File: packages/esm-patient-forms-app/src/clinical-form-action-button.test.tsx [hunk 2]
```diff
@@ -0,0 +1,26 @@
...
+
+describe('<ClinicalFormActionButton/>', () => {
+  test('should display clinical form action button on tablet view', () => {
+    spyOn(mockEsmFramework, 'useLayoutType').and.returnValue('tablet');
+    render(<ClinicalFormActionButton />);
+    const clinicalActionButton = screen.getByRole('button', { name: /Clinical form/ });
+
+    expect(clinicalActionButton).toHaveTextContent('Clinical form');
+    expect(clinicalActionButton).toBeInTheDocument();
+    expect(clinicalActionButton).toHaveClass('container');
```
Comment (brandones) [inline packages/esm-patient-forms-app/src/clinical-form-action-button.test.tsx:14]: Class names are implementation details, probably best not to test those.

File: packages/esm-patient-forms-app/src/clinical-form-action-button.test.tsx [hunk 3]
```diff
@@ -0,0 +1,26 @@
...
+import { render, screen } from '@testing-library/react';
+import * as mockEsmFramework from '@openmrs/esm-framework';
+
+describe('<ClinicalFormActionButton/>', () => {
+  test('should display clinical form action button on tablet view', () => {
+    spyOn(mockEsmFramework, 'useLayoutType').and.returnValue('tablet');
+    render(<ClinicalFormActionButton />);
+    const clinicalActionButton = screen.getByRole('button', { name: /Clinical form/ });
+
+    expect(clinicalActionButton).toHaveTextContent('Clinical form');
```
Comment (brandones) [inline packages/esm-patient-forms-app/src/clinical-form-action-button.test.tsx:12]: This is redundant with the `getByRole` call. I think the `expect...toBeInTheDocument` is enough.

File: packages/esm-patient-medications-app/src/medications-summary/order-basket-action-button.test.tsx [hunk 4]
```diff
@@ -0,0 +1,77 @@
...
+
+const mockedUseStore = useStore as jest.Mock;
+const mockedLayoutType = useLayoutType as jest.Mock;
+
+jest.mock('@openmrs/esm-framework', () => {
+  const originalModule = jest.requireActual('@openmrs/esm-framework');
+
+  return {
+    ...originalModule,
+    useStore: jest.fn(),
```
Comment (brandones) [inline packages/esm-patient-medications-app/src/medications-summary/order-basket-action-button.test.tsx:16]: `useStore` is in the esm-core mock

File: packages/esm-patient-medications-app/src/medications-summary/order-basket-action-button.test.tsx [hunk 5]
```diff
@@ -0,0 +1,77 @@
...
+    mockedUseStore.mockReturnValueOnce({ items: [] });
+    render(<OrderBasketActionButton />);
+
+    const orderBasketButton = screen.getByRole('button', { name: /Order Basket/i });
+    expect(orderBasketButton).toBeInTheDocument();
+    expect(orderBasketButton).toHaveClass('container');
+    userEvent.click(orderBasketButton);
+
+    expect(launchPatientWorkspace).toHaveBeenCalledWith('order-basket-workspace');
+    expect(orderBasketButton).toHaveClass('active');
```
Comment (brandones) [inline packages/esm-patient-medications-app/src/medications-summary/order-basket-action-button.test.tsx:44]: This is a useful class test, though :+1:

**Review by brandones — APPROVED** [review]

**Pull request merged**